src/Listeners/QueueNotificationJobs.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of fof/byobu.
+ *
+ * Copyright (c) 2019 FriendsOfFlarum.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FoF\Byobu\Listeners;
+
+use Flarum\Post\Event\Saving;
+use FoF\Byobu\Events\DiscussionMadePrivate;
+use FoF\Byobu\Jobs;
+use Illuminate\Events\Dispatcher;
+
+class QueueNotificationJobs
+{
+    public function subscribe(Dispatcher $events)
+    {
+        $events->listen(DiscussionMadePrivate::class, [$this, 'discussionMadePrivate']);
+        $events->listen(Saving::class, [$this, 'postMadeInPrivateDiscussion']);
+    }
+
+    public function discussionMadePrivate(DiscussionMadePrivate $event)
+    {
+        app('flarum.queue.connection')->push(
+            new Jobs\SendNotificationWhenPrivateDiscussionStarted($event->discussion, $event->newUsers)
+        );
+    }
+
+    public function postMadeInPrivateDiscussion(Saving $event)
+    {
+        $actor = $event->actor;
+
+        $event->post->afterSave(function ($post) use ($actor) {
+            if ($post->discussion->is_private && $post->number !== 1) {
```

Comment:
Not sure if I like this private check. What about other extensions that use private discussions (none right now, but imagine)?

I assume there's a way to check that the discussion is a byobu discussion, I think that'd be a better idea.

Reply:
I think, given that `is_private` is not a part of byobu specifically, we should consider some sort of token to be included on the `discussions` table, to indicate which extension 'owns' it.

Probably something for a later date?

Reply:
What about checking for recipients? Notify if there are any recipients & the post isn't the discussion's first post.

Reply:
Hmm, let me play around with that idea. `SendNotificationWhenPostedInPrivateDiscussion` does look there, but it makes sense to check here too.

Reply:
Ok, here's one solution...